django/db/models/query.py

```diff
@@ -61,7 +156,13 @@ def __init__(self, model=None, query=None, using=None, hints=None):
         self._for_write = False
         self._prefetch_related_lookups = []
```

Comment:
I'd like to get rid of `field_names`, `extra_names` and `annotation_names`, I believe it should be possible to get these directly for the `query`.